tests/conftest.py

```diff
@@ -2902,3 +2903,21 @@ def nmstate_namespace(admin_client):
 @pytest.fixture()
 def ipv6_single_stack_cluster(ipv4_supported_cluster, ipv6_supported_cluster):
     return ipv6_supported_cluster and not ipv4_supported_cluster
+
+
+@pytest.fixture(scope="class")
+def ping_process_in_rhel_os():
+    def _start_ping(vm):
+        return start_and_fetch_processid_on_linux_vm(
+            vm=vm,
+            process_name="ping",
+            args="localhost",
+        )
+
+    return _start_ping
```

Comment:
this doesn't seem right at all:
you're trying to make a fixture a callable in-test function which is total fixture logic break (fixture is triggered and executed in setup)

Reply:
https://stackoverflow.com/questions/51663326/why-would-a-pytest-factory-as-fixture-be-used-over-a-factory-function -- itried this approach and i ran virt tests and they passed with this approach

Reply:
thanks! just wasn't aware about this [factory usage of fixtures](https://docs.pytest.org/en/6.2.x/fixture.html#factories-as-fixtures)